packages/flame/lib/src/components/timer_component.dart

```diff
@@ -32,6 +38,18 @@ class TimerComponent extends Component {
     );
   }
 
+  @override
+  @mustCallSuper
+  FutureOr<void> onLoad() {
+    final result = super.onLoad();
+
+    if (tickOnLoad) {
+      onTick();
+    }
+
+    return result;
+  }
```

Comment:
```suggestion
  FutureOr<void> onLoad() async {
    await super.onLoad();

    if (tickOnLoad) {
      onTick();
    }
  }
```
We have to await it, otherwise we can't guarantee that it runs first.

Reply:
@spydon take another look please?